[Merged by Bors] - feat: Support arbitrary connectors using the generic connector (#436)
stackabletech/trino-operator: docs/modules/trino/pages/usage_guide/catalogs/generic.adoc

= Generic

> As Trino supports a lots of different connectors, we can't possible cover all the available connectors.

Reviewer: ```suggestion
Trino supports lots of different connectors and we can not cover all the available connectors.
```

> In case the trino-operator does not support a specific connector you can use this connector to pull in any connector Trino supports.

Reviewer: ```suggestion
In case the Trino operator does not support a specific connector you can use this generic connector to utilize any connector Trino supports.
```

> You are one you own figuring out how to configure and use this connector and we can't possible support all the catalogs available.

Reviewer: I dont like this. I would point to the Trino docs and e.g. mention that the connector name must match the one in the docs and that properties can be put here etc...
```suggestion
You are on your own figuring out how to configure and use this connector and we can't possible support all the catalogs available.
```

With this out of the way, this is how you can e.g. use the https://trino.io/docs/current/connector/postgresql.html[PostgreSQL connector]:

== Example generic catalog properties
[source,yaml]
----
apiVersion: trino.stackable.tech/v1alpha1
kind: TrinoCatalog
metadata:
  # The name of the catalog as it will appear in Trino
  name: postgres
  # TrinoCluster can use these labels to select which catalogs to include
  labels:
    trino: simple-trino
spec:
  connector:
    generic:
      connectorName: postgresql

>       properties: # optional

Reviewer: This is only optional for a few connectors right?

Author: Si, e.g. tpch, tpcds or the storage connector we use for a customer

        connection-url:
          value: jdbc:postgresql://example.net:5432/database
        connection-user:
          valueFromSecret:
            name: my-postgresql-credentials-secret
            key: user
        connection-password:
          valueFromSecret:
            name: my-postgresql-credentials-secret
            key: password
#         extra-property-from-configmap:
#           valueFromConfigMap:
#             name: my-configmap
#             key: my-key
---
apiVersion: v1
kind: Secret
metadata:
  name: my-postgresql-credentials-secret
stringData:
  user: root
  password: rootroot
----

== Add contents from Secrets or ConfigMaps
In some cases you don't want to add sensitive data (such as PostgreSQL credentials) to a `TrinoCatalog`.

> You can than use `valueFromSecret` or `valueFromSecret` to pull in properties securely.

Reviewer: ```suggestion
You can then use `valueFromSecret` or `valueFromSecret` to pull in properties securely.
```

> == Add contents from Secrets or ConfigMaps
> In some cases you don't want to add sensitive data (such as PostgreSQL credentials) to a `TrinoCatalog`.
> You can then use `valueFromSecret` or `valueFromSecret` to pull in properties securely.
> An entry must be an valid `SecretKeySelector` or `ConfigMapKeySelector`.
> 
> == Add additional contents
> In case you need more contents, such as the need to mount Secrets - containing e.g. certificates or keytabs - please consult the xref:nightly@home:concepts:overrides.adoc#_pod_overrides[documentation on podOverrides].

Reviewer: ```suggestion
== Add contents from Secrets or ConfigMaps

In some cases you don't want to add sensitive data (such as PostgreSQL credentials) to a `TrinoCatalog`.
You can then use `valueFromSecret` or `valueFromSecret` to pull in properties securely.
An entry must be an valid `SecretKeySelector` or `ConfigMapKeySelector`.

== Add additional contents

In case you need more contents, such as the need to mount Secrets - containing e.g. certificates or keytabs - please consult the xref:nightly@home:concepts:overrides.adoc#_pod_overrides[documentation on podOverrides].
```
